src/main/java/org/zalando/nakadi/repository/db/StorageDbRepository.java

```diff
@@ -46,11 +62,29 @@ public Storage createStorage(final Storage storage) throws DataAccessException {
         } catch (final JsonProcessingException ex) {
             throw new IllegalArgumentException("Storage configuration " + storage.getConfiguration(Object.class) +
                     " can't be mapped to json", ex);
+        } catch (DuplicateKeyException e) {
+            throw new DuplicatedStorageIdException("A storage with id '" + storage.getId() + "' already exists.", e);
+        } catch (DataAccessException e) {
+            throw new InternalNakadiException("Error occurred when creating Storage " + storage.getId(), e);
         }
     }
 
-    public void deleteStorage(final String id) {
-        jdbcTemplate.update("DELETE FROM zn_data.storage WHERE st_id=?", id);
+    public void deleteStorage(final String id) throws InternalNakadiException {
+        try {
+            jdbcTemplate.update("DELETE FROM zn_data.storage WHERE st_id=?", id);
+        } catch (DataAccessException e) {
+            throw new InternalNakadiException("Error occurred when deleting Storage " + id, e);
```

Comment:
I haven't found it to be logged somewhere. Probably, the logging of the error would be nice in the future.